unixfs/io/pbdagreader.go

```diff
@@ -51,19 +51,37 @@ type PBDagReader struct {
 var _ DagReader = (*PBDagReader)(nil)
 
 // NewPBFileReader constructs a new PBFileReader.
-func NewPBFileReader(ctx context.Context, n *mdag.ProtoNode, pb *ftpb.Data, serv ipld.NodeGetter) *PBDagReader {
-	fctx, cancel := context.WithCancel(ctx)
+func NewPBFileReader(ctx context.Context, n *mdag.ProtoNode, pb *ftpb.Data, serv ipld.NodeGetter) (*PBDagReader, error) {
 	curLinks := getLinkCids(n)
+	data := pb.GetData()
+
+	// validate
```

Comment:
I'd like to have this logic broken out separately and have a few unit tests for it

Reply:
Can you clarify what you mean here?

Reply:
This validate logic that has been newly added. It should live in its own function. And then we should add a few units tests for that function.

Reply:
Okay.

Reply:
Okay this is done now.  Sorry for the delay.